var/spack/repos/builtin/packages/trimmomatic/package.py

```diff
@@ -43,7 +43,7 @@ class Trimmomatic(Package):
 
     def install(self, spec, prefix):
         mkdirp(prefix.bin)
-        jar_file = 'trimmomatic-{}.jar'.format(self.version.dotted)
+        jar_file = 'trimmomatic-{v}.jar'.format(v=self.version.dotted)
```

Comment:
Yes, `{}` only works in Python 2.7+. To get things working in Python 2.6 you have to use `{0}` or `{variable}`.